### ISSUE TITLE
NumberBox does not respect Background

### ISSUE THREAD

**Issue by AndrewKeepCoding** [microsoft/microsoft-ui-xaml#9020]
### Describe the bug

This xaml:

```xaml
<StackPanel
    Orientation="Horizontal"
    Spacing="10">
    <TextBox
        Background="SkyBlue"
        Foreground="HotPink"
        Header="TextBox"
        Text="{x:Bind NumberBoxControl.Text, Mode=OneWay}" />
    <AutoSuggestBox
        Background="SkyBlue"
        Foreground="HotPink"
        Header="AutoSuggestBox"
        Text="{x:Bind NumberBoxControl.Text, Mode=OneWay}" />
    <NumberBox
        x:Name="NumberBoxControl"
        Background="SkyBlue"
        Foreground="HotPink"
        Header="NumberBox" />
</StackPanel>
```

renders:
![image](https://github.com/microsoft/microsoft-ui-xaml/assets/80524586/e1897652-445b-4197-add0-55000be5476f)

### Steps to reproduce the bug

Use the code above.

### Expected behavior

`NumberBox` respect its assigned `Background`.

### NuGet package version

WinUI 3 - Windows App SDK 1.4.2: 1.4.231008000

### Additional context

Related to:
- #2998 
- #4469

**Comment by AndrewKeepCoding** [comment]
According to [NumberBox.xaml](https://github.com/microsoft/microsoft-ui-xaml/blob/winui3/release/1.4-stable/controls/dev/NumberBox/NumberBox.xaml), the *InputBox* doesn't bring any `Background`.

**Comment by karkarl** [comment]
Fixed in internal PR: 10250839
